NeuroID/src/main/java/com/neuroid/tracker/utils/NIDMetaData.kt

```diff
@@ -82,6 +92,65 @@ class NIDMetaData(context: Context) {
         }
     }
 
+    internal fun getLocation(context: Context) {
+        if (ActivityCompat.checkSelfPermission(
+                context,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED || ActivityCompat.checkSelfPermission(
+                context,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            // need to request the permissions, return
+            return
+        }
+
+        // setup location manager
+        val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+
+        // get last position if available, take highest accuracy from available providers
+        val providers = locationManager.getProviders(true)
+        var smallestAccuracyMeters = Float.MAX_VALUE
+        for (provider in providers) {
+            val location = locationManager.getLastKnownLocation(provider)
+            if (location != null) {
+                if (location.accuracy < smallestAccuracyMeters) {
+                    gpsCoordinates.longitude = location.longitude
+                    gpsCoordinates.latitude = location.latitude
+                    gpsCoordinates.authorizationStatus = "authorizedAlways"
+                    smallestAccuracyMeters = location.accuracy
+                }
+            }
+        }
+
+        // unregister existing location listener, just in case
+        locationListener?.let {
+            locationManager.removeUpdates(it)
+        }
+
+        // setup new location listener, just in case
+        locationListener = LocationListener { location ->
+            gpsCoordinates.longitude = location.latitude
+            gpsCoordinates.latitude = location.longitude
+            gpsCoordinates.authorizationStatus = "authorizedAlways"
+        }
+
+        // register new listener, 1 minute min time interval and 10 meter min distance interval
+        // use GPS, highest accuracy
+        locationListener?.let {
+            locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER,
+                60000L, 10F, it)
+        }
+        NIDLog.d(tag = "TESTING", msg = "post check - ${this.gpsCoordinates}")
```

Comment:
I am assuming we are leaving this here until after the testing with DS?

Reply:
I intend to leave this in permanently since it only fires when app is the debug flavor

Reply:
I don't think that is the case anymore because we changed our logging system to work regardless of debug vs. production

Reply:
will remove now then.